vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx

```diff
@@ -45,108 +64,146 @@ module IntegersAndNumbers =
     printfn "%s" helloWorld
 
-    /// Substrings use the indexer notation, for example the first 7 characters.
+    /// Substrings use the indexer notation.  This line extracts the first 7 characters as a substring.
+    /// Note that like many languages, Strings are zero-indexed in F#.
     let substring = helloWorld.[0..6]
     printfn "%s" substring
 
 
-
 /// Tuples are simple combinations of data values into a combined value.
-module Tuples = 
+///
+/// To learn more, see: https://docs.microsoft.com/en-us/dotnet/articles/fsharp/language-reference/tuples
+module Tuples =
 
-    /// First define a simple tuple of integers.
+    /// A simple tuple of integers.
     let tuple1 = (1, 2, 3)
 
-    /// Next you define a function that swaps the order of two values in a tuple. 
-    /// The function is inferred to have a generic type.
+    /// A function that swaps the order of two values in a tuple. 
+    ///
+    /// F# Type Inference will automatically generalize the function to have a generic type,
+    /// meaning that it will work with any type.
     let swapElems (a, b) = (b, a)
 
     printfn "The result of swapping (1, 2) is %A" (swapElems (1,2))
 
-    /// A tuple consisting of an integer, a string, and a double-precision floating point number
+    /// A tuple consisting of an integer, a string,
+    /// and a double-precision floating point number.
     let tuple2 = (1, "fred", 3.1415)
 
-    printfn "tuple1: %A    tuple2: %A" tuple1 tuple2
+    printfn "tuple1: %A\ttuple2: %A" tuple1 tuple2
+
+    /// Tuples are normally objects, but they can also be represented as structs.
+    ///
+    /// These interoperate completely with structs in C# and Visual Basic.NET; however,
+    /// struct tuples are not implicitly convertable with object tuples (often called reference tuples).
+    ///
+    /// The second line below will fail to compile because of this.  Uncomment it to see what happens.
+    let sampleStructTuple = struct (1, 2)
+    //let thisWillNotCompile: (int*int) = struct (1, 2)
 
-    /// A simple tuple of integers, represented as a flat inline struct (a value-type)
-    let sampleStructTuple = struct (1, 2, 3)
+    // Although you can
+    let convertFromStructTuple (struct(a, b)) = (a, b)
+    let convertToStructTuple (a, b) = struct(a, b)
 
+    printfn "Struct Tuple: %A\nReference tuple made from the Struct Tuple: %A" sampleStructTuple (sampleStructTuple |> convertFromStructTuple)
 
-// ---------------------------------------------------------------
-//         Lists, Arrays and data processing
-// ---------------------------------------------------------------
 
-/// Lists, arrays and sequences are basic collection types in F# programming.
-/// Many other collection types are available in System.Collections.Generic,
-/// System.Collections.Immutable and other libraries.
-module ListsAndArrays = 
+/// Lists are ordered, immutable, singly-linked lists.  They are eager in their evaluation.
```

Comment:
immutability is a concept that is not used in many languages yet. Can we link to a definition?